Lib/test/test_asyncio/test_subprocess.py

```diff
@@ -686,6 +686,87 @@ async def execute():
 
         self.assertIsNone(self.loop.run_until_complete(execute()))
 
+    def test_create_subprocess_env_shell(self) -> None:
+        async def main() -> None:
+            cmd = f'''{sys.executable} -c "import os, sys; sys.stdout.write(os.getenv('FOO'))"'''
+            env = {"FOO": 'bar'}
+            proc = await asyncio.create_subprocess_shell(
+                cmd, env=env, stdout=subprocess.PIPE
+            )
+            stdout, _ = await proc.communicate()
+            self.assertEqual(stdout, b"bar")
+            self.assertEqual(proc.returncode, 0)
+            task = asyncio.create_task(proc.wait())
+            await asyncio.sleep(0)
+            self.assertEqual(task.result(), proc.returncode)
+
+        self.loop.run_until_complete(main())
+
+    def test_create_subprocess_env_exec(self) -> None:
+        async def main() -> None:
+            cmd = [sys.executable, "-c",
+                   "import os, sys; sys.stdout.write(os.getenv('FOO'))"]
+            env = {"FOO": 'bar'}
+            proc = await asyncio.create_subprocess_exec(
+                *cmd, env=env, stdout=subprocess.PIPE
+            )
+            stdout, _ = await proc.communicate()
+            self.assertEqual(stdout, b"bar")
+            self.assertEqual(proc.returncode, 0)
+            task = asyncio.create_task(proc.wait())
+            await asyncio.sleep(0)
+            self.assertEqual(task.result(), proc.returncode)
+
+        self.loop.run_until_complete(main())
```

Comment:
This block is the same as in the previous test -- maybe factor out into a helper function to reduce code duplication?

Reply:
done